src/lmdb.ml

```diff
@@ -330,48 +396,58 @@ module Make (Key : Values.S) (Elt : Values.S) = struct
       else def_flags
     in
     let f txn = mdb_dbi_open txn name flags db in
-    trivial_txn ~write:create env f ;
+    if create
+    then Txn.maybe rw env f
+    else Txn.maybe ro env f ;
+    { env ; db = !@db }
 
     (* We do not put a finaliser here, as it would break with mdb_drop.
        Slight memory leak, but nothing terrible. *)
     (* Gc.finalise mdb_dbi_close env !@db *)
-    { env ; db = !@db }
 
   let stats { env ; db } =
     let stats = make mdb_stat in
-    trivial_txn ~write:false env (fun t ->
+    Txn.maybe ro env (fun t ->
       mdb_dbi_stat t db (addr stats)
     ) ;
     Env.make_stats stats
 
   let _flags { env ; db } =
     let flags = alloc mdb_dbi_open_flag in
-    trivial_txn ~write:false env (fun t ->
+    Txn.maybe ro env (fun t ->
       mdb_dbi_flags t db flags
     ) ;
     !@flags
 
   let drop ?(delete=false) { env ; db } =
-    trivial_txn ~write:true env (fun t ->
+    Txn.maybe rw env (fun t ->
       mdb_drop t db delete
     )
```

Comment:
Do we want an optional transaction argument on create/stats/drop ? It seemed to me that was one of the purpose of the patch.

Reply:
Yes, we do and I have the code ready, but it does not _need_ to be done in this commit yet. Lets go one step at a time.